main.go

```diff
@@ -138,11 +152,53 @@ func walk(ch chan<- *file, start string) {
 }
 
 func addLicense(path string, fmode os.FileMode, tmpl *template.Template, data *copyrightData) (bool, error) {
+	var lic []byte
+	var err error
+	lic, err = getLicenseForExtension(path, tmpl, data)
+	if err != nil || lic == nil {
+		return false, err
+	}
+
+	b, err := ioutil.ReadFile(path)
+	if err != nil || hasLicense(b) {
+		return false, err
+	}
+
+	line := hashBang(b)
+	if len(line) > 0 {
+		b = b[len(line):]
+		if line[len(line)-1] != '\n' {
+			line = append(line, '\n')
+		}
+		lic = append(line, lic...)
+	}
+	b = append(lic, b...)
+	return true, ioutil.WriteFile(path, b, fmode)
+}
+
+func checkLicense(path string, tmpl *template.Template, data *copyrightData) (bool, error) {
```

Comment:
Let's name the return values so that it's clear what the bool means when "checking" a license:

```go
func checkFileLicense(...) (exists bool, err error)
```

Or, maybe even slightly better, how about expand on the existing `hasLicense` func and rename this to say:

```go
func fileHasLicense(path string, ...) (bool, error)
```

Then it's perfectly clear what the bool return values means.